### PR TITLE
[Github CI] Bump python to `3.9.5`

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR bumps Python to `3.9.5` for workflows still using `3.8` in the Github CI.

### Motivation

Have the Github CI consistent with the Gitlab CI. The [python version in the buildimages](https://github.com/DataDog/datadog-agent-buildimages/blob/a501d08b16e68292c5408e25028655219cc79390/setup_python.sh#L29) is `3.9.5`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]
LGTM for `agent-security` owned files

File: .github/workflows/cws-btfhub-sync.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       - name: Install python
         uses: actions/setup-python@v4
         with:
-          python-version: '3.8.14'
+          python-version: '3.9.5'
```
Comment (paulcacheux) [inline .github/workflows/cws-btfhub-sync.yml:31]: Maybe a quick question why 3.9.5 and not 3.9.18 (the latest at this time) ?

Comment (amenasria) [reply .github/workflows/cws-btfhub-sync.yml:31]: Because it's the version in the buildimages used by the CI (see `Motivation` section).

**Review by Pythyu — APPROVED** [review]
LGTM

**Pull request merged**